astropy/units/equivalencies.py

```diff
@@ -58,18 +56,21 @@ class Equivalency(UserList):
     """
 
     def __init__(self, equiv_list, name="", kwargs=None):
-        self.data = equiv_list
+        super().__init__(equiv_list)
         self.name = [name]
         self.kwargs = [kwargs] if kwargs is not None else [{}]
 
     def __add__(self, other):
         if isinstance(other, Equivalency):
-            new = super().__add__(other)
-            new.name = self.name[:] + other.name
-            new.kwargs = self.kwargs[:] + other.kwargs
+            # The super() returns a list, which is really a bit weird,
```

Comment:
Why would it be weird that `list.__add__()` returns a `list`?

Reply:
Usually if you call `super()`, you get back an instance of the subclass; I think the difference here is that `list.__new__(cls, ...)` returns a `list` rather than a `cls` instance (possibly because it is C code, where subclasses are, I think, a little trickier).